src/fsharp/FSharp.Core/array.fs

```diff
@@ -559,6 +559,31 @@ namespace Microsoft.FSharp.Collections
                     res.[i] <- Microsoft.FSharp.Primitives.Basics.Array.subUnchecked i windowSize array
                 res
 
+        [<CompiledName("ChunksOf")>]
+        let chunksOf chunkSize (array:'T[]) =
+            checkNonNull "array" array
+            if chunkSize <= 0 then invalidArg "chunkSize" (SR.GetString(SR.inputMustBeNonNegative))
```

Comment:
We need an error message `inputMustBePositive`, since 0 is non-negative but still invalid. From a quick scan, about 80% of usages for this message are correct (check for `< 0`) but a handful are inaccurate (check for `<= 0`).

Should not block this contribution, just something to look at later.